### PR TITLE
Added .form property for legend tag

### PR DESCRIPTION
**Pull request by autrilla**
r? @emilio

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10388)

### CODE REVIEW

File: components/script/dom/htmllegendelement.rs [hunk 1]
```diff
@@ -33,3 +41,45 @@ impl HTMLLegendElement {
...
+        } else {
+            el.check_disabled_attribute();
+        }
+    }
+}
+
+
+impl HTMLLegendElementMethods for HTMLLegendElement {
+    fn GetForm(&self) -> Option<Root<HTMLFormElement>> {
+		let parent = match self.upcast::<Node>().GetParentElement() {
```
Comment (KiChjang) [inline components/script/dom/htmllegendelement.rs]: nit: Tabs on this line, this is going to fail tidy.

File: components/script/dom/htmllegendelement.rs [hunk 2]
```diff
@@ -33,3 +41,45 @@ impl HTMLLegendElement {
...
+            Some(parent) => parent,
+            None => return None,
+		};
+		if parent.TagName() == "FIELDSET" {
+			return self.form_owner();
+		}
+		None
```
Comment (KiChjang) [inline components/script/dom/htmllegendelement.rs]: Tabs on the above 5 lines.

**Comment by emilio** [comment]
Looks great! Just some nits.

Let's give this a try run since I'm sure a few test expectations will need to change :)

@bors-servo: try

---

Reviewed 2 of 3 files at r1.
Review status: 2 of 3 files reviewed at latest revision, 3 unresolved discussions.

---

*[components/script/dom/htmllegendelement.rs, line 72 \[r2\]](https://reviewable.io:443/reviews/servo/servo/10388#-KESM5Hr0J2yrcrGU0aF:-KESM5Hr0J2yrcrGU0aG:1696718469) ([raw file](https://github.com/servo/servo/blob/599bcdd0e5357dd9d54202a3d2caf9e18290c982/components/script/dom/htmllegendelement.rs#L72)):*
nit: This should point to the `form` attribute, that is: https://html.spec.whatwg.org/multipage/#dom-legend-form

---

*[components/script/dom/htmllegendelement.rs, line 78 \[r2\]](https://reviewable.io:443/reviews/servo/servo/10388#-KESKZ_igT2lzDm-Z8-G:-KESKZ_igT2lzDm-Z8-H:137998265) ([raw file](https://github.com/servo/servo/blob/599bcdd0e5357dd9d54202a3d2caf9e18290c982/components/script/dom/htmllegendelement.rs#L78)):*
nit: The proper way to do this is using `parent.is::<HTMLFieldsetElement>`.

---

*[tests/wpt/web-platform-tests/html/semantics/forms/the-legend-element/legend-form.html, line 26 \[r2\]](https://reviewable.io:443/reviews/servo/servo/10388#-KESKKKbvCCokq1w6dr_:-KESKKKbvCCokq1w6dra:-646151588) ([raw file](https://github.com/servo/servo/blob/599bcdd0e5357dd9d54202a3d2caf9e18290c982/tests/wpt/web-platform-tests/html/semantics/forms/the-legend-element/legend-form.html#L26)):*
Add an extra line under this like:

```js
assert_equals(legendEle.form, document.getElementById("testformWithFieldSet"));
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/10388#-:-KESLhwDm9OgMNowyBtk:1317462401)*

**Comment by emilio** [comment]
-S-awaiting-review +S-needs-code-changes

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/10388#-:-KESMeL8Bi7JnfR6Apg3:324135503)*

**Comment by emilio** [comment]
@bors-servo: try

 * bors was dead

**Comment by jdm** [comment]
@bors-servo: retry
* weird git error

**Comment by jdm** [comment]
@bors-servo: retry
* more weird git errors

**Comment by KiChjang** [comment]
```
Ran 4648 tests finished in 298.0 seconds.
  • 4646 ran as expected. 1364 tests skipped.
  • 2 tests had unexpected subtest results

Tests with unexpected results:
  ▶ Unexpected subtest result in /html/dom/interfaces.html:
  └ PASS [expected FAIL] HTMLLegendElement interface: attribute form

  ▶ Unexpected subtest result in /html/dom/interfaces.html:
  └ PASS [expected FAIL] HTMLLegendElement interface: document.createElement("legend") must inherit property "form" with the proper type (0)

  ▶ Unexpected subtest result in /html/semantics/forms/the-legend-element/legend-form.html:
  └ PASS [expected FAIL] Check if legend.form return null when legend has no fieldset element as its parent
```

**Comment by emilio** [comment]
Yay!

@autrilla You'll need to update the test expectations to mark those tests as passing, let us know if you need any help with it :P

**Comment by autrilla** [comment]
@emilio done :)

**Comment by emilio** [comment]
Just some nits left! Also, may you squash the commits into one? :)

-S-awaiting-review +S-needs-code-changes 

---

Review status: 1 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[tests/wpt/metadata/html/dom/interfaces.html.ini, line 5322 \[r4\]](https://reviewable.io:443/reviews/servo/servo/10388#-KEfrhTLzetXy3F0mdLI:-KEfrhTMXYiTQHREOuRp:678576188) ([raw file](https://github.com/servo/servo/blob/9bef4df8f3f8c8e573d9f639a4c3a787e54c30f4/tests/wpt/metadata/html/dom/interfaces.html.ini#L5322)):*
Pass is the default, so the ini sections should be removed :)

---

*[tests/wpt/metadata/html/semantics/forms/the-legend-element/legend-form.html.ini, line 4 \[r4\]](https://reviewable.io:443/reviews/servo/servo/10388#-KEfrlVumNX3KzjkrPGY:-KEfrlVumNX3KzjkrPGZ:124778897) ([raw file](https://github.com/servo/servo/blob/9bef4df8f3f8c8e573d9f639a4c3a787e54c30f4/tests/wpt/metadata/html/semantics/forms/the-legend-element/legend-form.html.ini#L4)):*
Same here, you should remove the whole file.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/10388#-:-KEfrugrTRAvpIZ1MQ7t:-461972590)*

**Comment by autrilla** [comment]
@emilio changed and squashed!

**Pull request merged**